bin/utils/component-utils.sh

```diff
@@ -503,3 +503,175 @@ process_component_appfw_plugin() {
+# it has any desktop plugins it will check if it exists on the zowe instance.
+#
+# Note: this function calls is dependent on various utility functions and
+#       environment variables. Simply source bin/internal/prepare-environment.sh
+#       to import the required dependencies.
+#
+# Required environment variables:
+# - NODE_HOME
+# - ROOT_DIR
+# - ZOWE_EXPLORER_HOST
+# - DISCOVERY_PORT
+# - ZOWE_ZLUX_SERVER_HTTPS_PORT
+#
+# Example:
+# - Verify service and desktop plugin of explorer-jes component
+#   i.e: verify_component_instance "${ROOT_DIR}/components/explorer-jes"
+#
+# @param string   component directory
+verify_component_instance() {
+  component_id=$1
+
+  component_dir=$(find_component_directory "${component_id}")
+
+  service_ids=$(list_component_service_id "${component_dir}")
+
+  if [ ! -d ~/temp-verify ]; then
+    mkdir ~/temp-verify
+  fi
+
+  echo ${component_id} | tr '[:lower:]' '[:upper:]'
+  echo "==========================================\n"
+
+  echo "SERVICE COMPONENT STATUS"
+
+  for service_id in $service_ids; do
+    rm -rf ~/temp-verify/verify-ext-serviceid.json
+    node "${ROOT_DIR}"/bin/utils/curl.js https://"${ZOWE_EXPLORER_HOST}":"${DISCOVERY_PORT}"/eureka/apps/"${service_id}" -k -H 'Accept: application/json' -J >> ~/temp-verify/verify-ext-serviceid.json
+    service_status=$(read_json ~/temp-verify/verify-ext-serviceid.json .application.instance[0].status)
+    echo "${service_id}" - "${service_status}"
+  done
+
+  if [ -z "${service_ids}" ]; then
+     echo "No services exist for this component"
+  fi
+  
+  desktop_ids=$(list_component_plugin_id "${component_dir}")
+  
+  echo "\nDESKTOP COMPONENT STATUS"
+
+  for desktop_id in $desktop_ids; do
+    rm -rf ~/temp-verify/verify-ext-desktop.json
+    node "${ROOT_DIR}"/bin/utils/curl.js https://"${ZOWE_EXPLORER_HOST}":"${ZOWE_ZLUX_SERVER_HTTPS_PORT}"/plugins -k >> ~/temp-verify/verify-ext-desktop.json
+    desktop_identifier_list=$(read_json ~/temp-verify/verify-ext-desktop.json .pluginDefinitions[].identifier)
+
+    if [[ "$desktop_identifier_list" == *"$desktop_id"* ]]; then
+      echo "OK! - ${desktop_id} exists"
+    else
+      echo "FAIL! - ${desktop_id} does not exist"
+    fi
+  done
+
+  if [ -z "${desktop_ids}" ]; then
+    echo "No desktop plugins exist for this component"
+  fi
+}
+
+###############################
+# Lists all the components that exists in a zowe instance
+#
+# Required environment variables:
+# - ROOT_DIR
+# - ZWE_EXTENSION_DIR
+#
+# Example:
+# - This will display all components that is currentally installled on a zowe instance
+#   i.e: list_all_components
+#
+# @param string   component directory
+list_all_components() {
+  # Take note: find_component_directory doesn't locate {{ zwe_extensions_dir }}
+  # temporary, need to change variable
+  component_dir_list="${ROOT_DIR}/components ${ROOT_DIR}/../extensions"
+  for component_dirs in ${component_dir_list}; do
+    ls -1 ${component_dirs}
+  done
+}
```

Comment:
@jackjia-ibm. Concerned about the list_all_components() function, we've discussed about placing the component directory paths into one array. I wasn't sure on how to retrieve the zwe_extensions_dir so I used a relative path from ${ROOT_DIR} for now.

Reply:
For now, I used the ZWE_EXTENSION_DIR env variable.